File: backend/app/schemas/signal.py
# backend/app/schemas/signal.py

from datetime import datetime
from typing import Optional
from app.schemas.base import BaseResponse
from app.schemas.trade import SecurityInfo


class SignalResponse(BaseResponse):
    id: int
    security: SecurityInfo
    observed_at: datetime
    payload: dict
    strategy_run_id: int
    strategy_name: Optional[str] = None
    trade_id: Optional[int] = None
    trade_status: Optional[str] = None
    trade_fill_price: Optional[float] = None
    trade_entry_date: Optional[str] = None
    trade_pnl_pct: Optional[float] = None
    trade_pnl: Optional[float] = None
    signal_close: Optional[float] = None
    latest_close: Optional[float] = None
    perf_since_signal: Optional[float] = None
    signal_status: str

File: frontend/libraries/types/signal.ts
import { Security } from "./trade"

export interface Signal {
    id: number
    security: Security
    observed_at: string
    payload: Record<string, unknown>
    strategy_run_id: number
    trade_id: number | null
    trade_status: string | null
    trade_fill_price: number | null
    trade_entry_date: string | null
    signal_status: "entered" | "missed"
}
